file/builder.go

```diff
@@ -771,9 +778,38 @@ func (b *stateBuilder) ingestRoute(r FRoute) error {
 	return b.ingestPlugins(plugins)
 }
 
+func (b *stateBuilder) getPluginSchema(pluginName string) (map[string]interface{}, error) {
+	var schema map[string]interface{}
+
+	// lookup in cache
+	if schema, ok := b.schemasCache[pluginName]; ok {
+		return schema, nil
+	}
+	schema, err := b.client.Plugins.GetFullSchema(b.ctx, &pluginName)
+	if err != nil {
+		return schema, err
+	}
+	b.schemasCache[pluginName] = schema
+	return schema, nil
+}
+
+func (b *stateBuilder) ingestPluginDefaults(plugin *FPlugin) error {
```

Comment:
```suggestion
func (b *stateBuilder) addPluginDefaults(plugin *FPlugin) error {
```
"ingest" has a slightly different meaning in this file it seems, more about "ingesting the objects into state".

Reply:
done